dask/dataframe/utils.py

```diff
@@ -432,41 +432,42 @@ def meta_nonempty(x):
 ###############################################################
 
 
-def _check_dask(dsk, check_names=True, check_dtypes=True):
+def _check_dask(dsk, check_names=True, check_dtypes=True, result=None):
     import dask.dataframe as dd
     if hasattr(dsk, 'dask'):
-        result = dsk.compute(get=get_sync)
+        if result is None:
+            result = dsk.compute(get=get_sync)
         if isinstance(dsk, dd.Index):
             assert isinstance(result, pd.Index), type(result)
-            if check_names:
-                assert dsk.name == result.name
-            # cache
             assert isinstance(dsk._meta, pd.Index), type(dsk._meta)
             if check_names:
+                assert dsk.name == result.name
```

Comment:
This is minor, but all index objects should have a `.names`, so you could always check `dsk.names == result.names` and avoid the `isinstance(., pd.MultiIndex)` check. For non-MI it should just be `FrozenList([dsk.name])`